casper/src/test/resources/PoSTest.rho

```diff
@@ -3,32 +3,290 @@
+                for (@(true, vault) <- vaultCh) {
+                  retCh!(pk.hexToBytes(), revAddress, vault)
+                }
+              }
+            }
+          } |
+
+          contract test_make_pos_succeeds(rhoSpec, _, ackCh) = {
+            new retCh in {
+              @PoS!("getBonds", *retCh) |
+              for ( @bonds <- retCh) {
+                rhoSpec!("assert",
+                  bonds.size() >= 0,
+                  "some bonds map should be received",
+                  *ackCh
+                )
+              }
+            }
+          } |
+
+          contract test_bonding_succeeds(rhoSpec, _, ackCh) = {
+            new setupCh, retCh, initialPosBalanceCh, finalPosBalanceCh,
+                initialUserBalanceCh, finalUserBalanceCh in {
+              prepareUser!("1111111111111111111111111111111111111111111111111111111111111111", 10000, *setupCh) |
+              for (@user1PubKey, _, @user1Vault <- setupCh) {
+                @posVault!("balance", *initialPosBalanceCh) |
+                @user1Vault!("balance", *initialUserBalanceCh) |
+                for(@initialPosBalance <- initialPosBalanceCh;
+                    @initialUserBalance <- initialUserBalanceCh) {
+                  @PoS!("bond", 100, *retCh) |
+                  for ( @(true, _) <- retCh) {
+                    @PoS!("getBonds", *retCh) |
+                    @posVault!("balance", *finalPosBalanceCh) |
+                    @user1Vault!("balance", *finalUserBalanceCh) |
+                    for(@finalPosBalance <- finalPosBalanceCh;
+                        @finalUserBalance <- finalUserBalanceCh;
+                        @bonds <- retCh) {
+                      stdlog!("info", {"finalPosBalance": finalPosBalance, "finalUserBalance":finalUserBalance, "bonds":bonds}) |
+                      match bonds.get(user1PubKey) {
+                        [bondAmount, _] => {
+                          rhoSpec!("assertMany",
+                            [
+                              ((100, "==", bondAmount), "the new bond is expected in the final map"),
+                              ((initialUserBalance - finalUserBalance, "==", 100), "the user account decreases"),
+                              ((finalPosBalance - initialPosBalance, "==", 100), "the pos account increases"),
+                            ],
+                            *ackCh
+                          )
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          } |
+
+          contract test_multiple_bonding_succeeds(rhoSpec, _, ackCh) = {
+            new setupCh, retCh, initialPosBalanceCh, finalPosBalanceCh,
+                initialUser1BalanceCh, finalUser1BalanceCh,
+                initialUser2BalanceCh, finalUser2BalanceCh in {
+              prepareUser!("2222222222222222222222222222222222222222222222222222222222222222", 10000, *setupCh) |
```

Comment:
having those looooong literals replaced with variables ("constatns") by using a `match` on the top would help

Reply:
Not really. They are only used once. and then I would have user1PublicKey, user2PublicKey... user8PublicKey. Not fun! 

This way they are only used in one single place i.e. the call to prepareUser

Reply:
I guess it's a matter of preference until they're repeated